src/rules/sort-collections.ts

```diff
@@ -1,125 +1,148 @@
+					nodeKey.type !== "JSONLiteral" ||
+					collection.type !== "JSONObjectExpression"
+				) {
+					return;
+				}
+
+				const keyPartsReversed = [nodeKey.value];
+				for (
+					let currNode: JsonAST.JSONNode | null | undefined =
+						node.parent;
+					// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+					currNode;
+					currNode = currNode.parent
 				) {
-					const currentOrder = collection.properties;
-					const scripts = new Set(
-						currentOrder.map(
-							(prop) => (prop.key as AST.JSONStringLiteral).value,
-						),
-					);
-
-					const desiredOrder = currentOrder.slice().sort((a, b) => {
-						let aKey = (a.key as AST.JSONStringLiteral).value;
-						let bKey = (b.key as AST.JSONStringLiteral).value;
-
-						if (key.value !== "scripts") {
-							return aKey > bKey ? 1 : -1;
-						} else {
-							let modifier = 0;
-
-							if (
-								aKey.startsWith("pre") &&
-								scripts.has(aKey.substring(3))
-							) {
-								aKey = aKey.substring(3);
-								modifier -= 1;
-							} else if (
-								aKey.startsWith("post") &&
-								scripts.has(aKey.substring(4))
-							) {
-								aKey = aKey.substring(4);
-								modifier += 1;
-							}
-
-							if (
-								bKey.startsWith("pre") &&
-								scripts.has(bKey.substring(3))
-							) {
-								bKey = bKey.substring(3);
-								modifier += 1;
-							} else if (
-								bKey.startsWith("post") &&
-								scripts.has(bKey.substring(4))
-							) {
-								bKey = bKey.substring(4);
-								modifier -= 1;
-							}
-
-							if (aKey === bKey) {
-								return modifier;
-							}
-
-							return aKey > bKey ? 1 : -1;
-						}
-					});
 					if (
-						currentOrder.some(
-							(property, i) => desiredOrder[i] !== property,
-						)
+						currNode.type === "JSONProperty" &&
+						currNode.key.type === "JSONLiteral"
 					) {
-						context.report({
-							data: {
-								key: key.value,
-							},
-							fix(fixer) {
-								return fixer.replaceText(
-									collection as unknown as ESTree.Node,
-									JSON.stringify(
-										desiredOrder.reduce<
-											Record<string, unknown>
-										>((out, property) => {
-											out[
-												(
-													property.key as AST.JSONStringLiteral
-												).value
-											] = JSON.parse(
-												context.sourceCode.getText(
-													property.value as unknown as ESTree.Node,
-												),
-											);
-											return out;
-										}, {}),
-										null,
-										2,
-									)
-										.split("\n")
-										.join("\n  "), // nest indents
-								);
-							},
-							loc: collection.loc,
-							messageId: "notAlphabetized",
-							node: node as unknown as ESTree.Node,
-						});
+						keyPartsReversed.push(currNode.key.value);
 					}
 				}
+				const key = keyPartsReversed.reverse().join(".");
+				if (!toSort.has(key)) {
+					return;
+				}
+
+				const currentOrder = collection.properties;
+				const scripts = new Set(
```

Comment:
Nit: I think is probably a holdover from when this was just sorting scripts.  Can we rename this to something more in line with its current purpose?  `properties` maybe?